### PR TITLE
[Merged by Bors] - feat: port Algebra.Category.GroupCat.EpiMono

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #3861
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3861
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Algebra/Category/GroupCat/EpiMono.lean [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+! leanprover-community/mathlib commit 70fd9563a21e7b963887c9360bd29b2393e6225a
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Category.GroupCat.EquivalenceGroupAddGroup
+import Mathlib.GroupTheory.QuotientGroup
+
+/-!
+# Monomorphisms and epimorphisms in `Group`
+In this file, we prove monomorphisms in category of group are injective homomorphisms and
```
Comment (semorrison) [inline Mathlib/Algebra/Category/GroupCat/EpiMono.lean:16]:
```suggestion
In this file, we prove monomorphisms in the category of groups are injective homomorphisms and
```

**Comment by semorrison** [comment]
bors d+

Please wait until the dependent PR lands. The issue with instances for `X.\a` is annoying, but there's no point holding up this PR for that.

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3861~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Algebra/Category/GroupCat/EpiMono.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/Algebra/Category/GroupCat/EpiMono.lean]: Added

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**